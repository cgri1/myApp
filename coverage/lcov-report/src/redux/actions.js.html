<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/redux/actions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/redux</a> actions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/4</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { store } from "./store";
&nbsp;
export function <span class="fstat-no" title="function not covered" >setData(</span>data) {
<span class="cstat-no" title="statement not covered" >  store.dispatch({</span>
    type: "SET_DATA",
    payload: {
      data
    }
  });
}
&nbsp;
&nbsp;
&nbsp;
export function <span class="fstat-no" title="function not covered" >setStoryList(</span>data) {
<span class="cstat-no" title="statement not covered" >  store.dispatch({</span>
    type: "SET_STORY_LIST",
    payload: {
      data
    }
  });
}
&nbsp;
export function <span class="fstat-no" title="function not covered" >setVoteStatus(</span>data) {
<span class="cstat-no" title="statement not covered" >  store.dispatch({</span>
    type: "SET_VOTE_STATUS",
    payload: {
      data
    }
  });
}
&nbsp;
export function <span class="fstat-no" title="function not covered" >setRoute(</span>flag) {
<span class="cstat-no" title="statement not covered" >  store.dispatch({</span>
    type: "SET_ROUTE",
    payload: {
      flag
    }
  });
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Mar 23 2020 10:17:10 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
