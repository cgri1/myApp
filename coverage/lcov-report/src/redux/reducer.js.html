<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/redux/reducer.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/redux</a> reducer.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">55.56% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">28.57% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">55.56% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/9</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import assign from "lodash/assign";
&nbsp;
const defaultValues = {
  data: [],
  storyList:[],
  votedList: [],
  flag: false
};
&nbsp;
export const setData = function (
  state = defaultValues,
  action = <span class="branch-0 cbranch-no" title="branch not covered" >{ type: "UNKNOWN" }</span>
) {
  const { payload } = action;
  switch (action.type) {
<span class="branch-0 cbranch-no" title="branch not covered" >    case "SET_DATA":</span>
<span class="cstat-no" title="statement not covered" >      return assign({}, state, {</span>
        data: payload.data
      });
<span class="branch-1 cbranch-no" title="branch not covered" >    case "SET_STORY_LIST":</span>
<span class="cstat-no" title="statement not covered" >      return assign({}, state, {</span>
        storyList: payload.data
      });
<span class="branch-2 cbranch-no" title="branch not covered" >    case "SET_VOTE_STATUS":</span>
<span class="cstat-no" title="statement not covered" >      return assign({}, state, {</span>
        votedList: payload.data
      });
<span class="branch-3 cbranch-no" title="branch not covered" >    case "SET_ROUTE":</span>
<span class="cstat-no" title="statement not covered" >      return assign({}, state, {</span>
        flag: payload.flag
      });
    default:
      return state;
  }
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Mar 23 2020 10:17:10 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
