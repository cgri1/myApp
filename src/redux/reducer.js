import assign from "lodash/assign";

const defaultValues = {
  data: [],
  storyList:[],
  votedList: [],
  flag: false
};

export const setData = function (
  state = defaultValues,
  action = { type: "UNKNOWN" }
) {
  const { payload } = action;
  switch (action.type) {
    case "SET_DATA":
      return assign({}, state, {
        data: payload.data
      });
    case "SET_STORY_LIST":
      return assign({}, state, {
        storyList: payload.data
      });
    case "SET_VOTE_STATUS":
      return assign({}, state, {
        votedList: payload.data
      });
    case "SET_ROUTE":
      return assign({}, state, {
        flag: payload.flag
      });
    default:
      return state;
  }
};
