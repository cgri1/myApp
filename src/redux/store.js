import { createStore } from "redux";
import { setData } from "./reducer";
import { combineReducers } from "redux";

const reducercombined = combineReducers({ setData });

export const store = createStore(reducercombined);
