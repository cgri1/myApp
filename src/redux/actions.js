import { store } from "./store";

export function setData(data) {
  store.dispatch({
    type: "SET_DATA",
    payload: {
      data
    }
  });
}



export function setStoryList(data) {
  store.dispatch({
    type: "SET_STORY_LIST",
    payload: {
      data
    }
  });
}

export function setVoteStatus(data) {
  store.dispatch({
    type: "SET_VOTE_STATUS",
    payload: {
      data
    }
  });
}

export function setRoute(flag) {
  store.dispatch({
    type: "SET_ROUTE",
    payload: {
      flag
    }
  });
}
