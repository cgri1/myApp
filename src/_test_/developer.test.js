import React from "react";
import Developer from "../components/developer";
import { shallow, render, mount } from "enzyme";
import { Provider } from "react-redux";
import { store } from "../redux/store";
// Create the mock store
import configureMockStore from "redux-mock-store";
const initialState = {};
const mockStore = configureMockStore(initialState);

test("Developer snapshot", () => {
  const component = shallow(
    <Provider store={store}>
      <Developer />
    </Provider>
  );
  expect(component).toMatchSnapshot();
});
