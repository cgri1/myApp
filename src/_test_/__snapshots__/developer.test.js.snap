// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Developer snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "setData": Object {
          "data": Array [],
          "flag": false,
          "storyList": Array [],
          "votedList": Array [],
        },
      },
    }
  }
>
  <withRouter(Connect(Developer)) />
</ContextProvider>
`;
